### ISSUE TITLE
Fwd: gevent.local is broken on PyPy2-7.3.3

### ISSUE THREAD

**Issue by navytux** [gevent/gevent#1709]
* gevent version: 20.9.0
* Python version: PyPy 7.3.3 - Python 2.7.18 (pypy2.7-v7.3.3-linux64.tar.bz2 downloaded from pypy.org)
* Operating System: `Linux deco 5.9.0-3-amd64 #1 SMP Debian 5.9.9-1 (2020-11-19) x86_64 GNU/Linux`

### Description:

Testing pygolang on pypy revealed the following:

```
>>>> gevent.local.local()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unbound method __new__() must be called with local instance as first argument (got type instance instead)
```

It used to work with PyPy 7.3.2:

https://foss.heptapod.net/pypy/pypy/-/issues/3352#note_148448

**Comment by navytux** [comment]
It looks like the problem is on gevent side:

https://foss.heptapod.net/pypy/pypy/-/issues/3352#note_148577  
https://foss.heptapod.net/pypy/pypy/-/issues/3352#note_148623

**Comment by jamadden** [comment]
Fixed in 1905f56d9e1a481b49cbffbeae7a54c662769bba.

Thanks for the report.

**Comment by navytux** [comment]
@jamadden you are welcome. Thanks for the fix.
